runtime/vm/jvminit.c

```diff
@@ -2392,9 +2393,10 @@ static UDATA checkArgsConsumed(J9PortLibrary* portLibrary, J9VMInitArgs* j9vm_ar
 		if (IS_CONSUMABLE( j9vm_args, i ) && !IS_CONSUMED( j9vm_args, i )) {
 			char* optString = j9vm_args->actualVMArgs->options[i].optionString;
 			char* envVar = j9vm_args->j9Options[i].fromEnvVar;
-
-			/* If ignoreUnrecognized is set to JNI_TRUE, we should not reject any unrecognized options beginning with -X or _ */
-			if (ignoreUnrecognized && optString && (!strncmp(optString, "-X", 2) || *optString=='_')) {
+			
+			/* If ignoreUnrecognized is set to JNI_TRUE, we should not reject any options that are: 
+				empty or contain only whitespace, or unrecognized options beginning with -X or _ */
+			if (ignoreUnrecognized && (NULL != optString) && (isEmpty(optString) || !strncmp(optString, "-X", 2) || *optString=='_')) {
```

Comment:
This doesn't ignore `NULL`s which I thought was part of the goal (to avoid segfaults).

Reply:
This was never the goal. Perhaps we should deal with that separately from this PR. Referring back to an earlier comment OpenJ9 and the RI have the same behavior `adding a NULL string to the args array options[0].optionString = NULL; results in a seg fault for both the RI and the J9 implementations.`

Reply:
Perhaps that better left alone (or at least for another day). It might even be considered helpful to identify problems in the application attempting to launch a VM.